BatteryChecker.c

```diff
@@ -0,0 +1,50 @@
+#include <stdio.h>
+#include <assert.h>
+#include "BatteryChecker.h"
+
+tst_BatteryStatus batteryStatus_st;
+
+int BatteryParamInRange(ten_BatteryParameter parameter, float value, float min, float max){
+	return ((value >= min) && (value <= max));
+}
+
+int CheckBatteryPerformance(ten_BatteryParameter parameter, float value, float min, float max){
+	int paramSts = 0, warningSts = 0;
+	paramSts = BatteryParamInRange(parameter, value, min, max);
+	if(paramSts == 0){
+		printf("\nParameter is out of range");
+		UpdateForLowOrHighBreach(parameter, value, min, max);
+	}
+	else{
+		warningSts = CheckForWarning(parameter, value, min, max);
+	}
+	return paramSts && !warningSts;
+}
+
+int batteryIsOk(float temperature, ten_TempUnit tempUnit, float soc, float chargeRate) {
+	float temperatureInCelcius = 0;
+	temperatureInCelcius = TempConversion(temperature, tempUnit);
+	return (CheckBatteryPerformance(TEMPERATURE, temperatureInCelcius, TEMP_MIN, TEMP_MAX) && CheckBatteryPerformance(SOC, soc, SOC_MIN, SOC_MAX) &&
+			CheckBatteryPerformance(CHARGERATE, chargeRate, CHARGERATE_MIN, CHARGERATE_MAX));
+}
+
+int main(void) {
+  assert(batteryIsOk(25, CELCIUS, 70, 0.7)); 		//All params in Range
```

Comment:
Exhaustive tests 👍 
Evaluate if the number of tests gets reduced if you test the abstractions (`Update...For...` functions) separately.